### PR TITLE
M3-714 Profile: Update email address

### PR DESCRIPTION
**Pull request by Jskobos**
## Purpose

Users should be able to update their account email address when viewing their profile.

* Add form for updating email/username
* Add routing logic to Profile.tsx
* Add DisplaySettings component (will later hold timezone as well)
* Add tests for 2 new components

## Notes

* The original design included an editable username field as well. Since this is not supported by the CF manager or the /profile endpoint, this has been dropped. The textfield with the user's display name is disabled, it may need to be removed pending design updates or review.

### CODE REVIEW

File: src/features/profile/DisplaySettings/EmailChangeForm.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        <React.Fragment>
+            <TextField
+                disabled
+                label="Username"
+                value={username}
+                errorGroup="display-settings-email"
+                data-qa-username
+            />
+            <TextField
+                label="Email"
```
Comment (martinmckenna) [inline src/features/profile/DisplaySettings/EmailChangeForm.tsx]: I would give this a `type` prop of `email` so mobile uses the email keyboard

File: src/features/profile/DisplaySettings/EmailChangeForm.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                value={email}
+                onChange={handleChange}
+                errorText={error}
+                errorGroup="display-settings-email"
+                error={Boolean(error)}
```
Comment (rjstires) [inline src/features/profile/DisplaySettings/EmailChangeForm.tsx]: IIRC we don't need to send in error explicitly to TextField as the errorText handles that for us.

File: src/features/profile/DisplaySettings/EmailChangeForm.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            submitting,
+            username, } = props;
+    return (
+        <React.Fragment>
+            <TextField
+                disabled
+                label="Username"
+                value={username}
+                errorGroup="display-settings-email"
+                data-qa-username
```
Comment (patthiel) [inline src/features/profile/DisplaySettings/EmailChangeForm.tsx]: thank you for adding these 😄

**Comment by patthiel** [comment]
Will retest after @rjstires & @martinmckenna changes are addressed. Things looked good when i tested though

**Comment by asauber** [comment]
@Jskobos I agree with @rjstires, the state management should be moved into the component

File: src/features/profile/DisplaySettings/EmailChangeForm.tsx (hunk 2 above)
Comment (asauber) [reply src/features/profile/DisplaySettings/EmailChangeForm.tsx]: that's right

File: src/features/profile/DisplaySettings/DisplaySettings.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    const emailError = hasErrorFor('email');
+    return (
+      <React.Fragment>
+        <Paper className={classes.root}>
+          <Typography
+            variant="title"
+            className={classes.title}
+            data-qa-title
+          >
+            Email Address
```
Comment (asauber) [inline src/features/profile/DisplaySettings/DisplaySettings.tsx]: This will be seen as a "duplicate label". It should probably be removed.

**Comment by asauber** [comment]
The indentation for EmailChangeForm is 4 spaces, should be 2

**Pull request merged**